packages/Search/src/details/DTK_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -258,38 +256,46 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
     ////////////////////////////////////////////////////////////////////////////
 }
 
+// NOTE: for some reason Kokkos::BinSort::sort() is not const...
+template <typename BinSort>
+void applyPermutations( BinSort & )
+{
+    // do nothing
+}
+
+template <typename BinSort, typename View, typename... OtherViews>
+void applyPermutations( BinSort &bin_sort, View view,
+                        OtherViews... other_views )
+{
+    bin_sort.sort( view );
+    applyPermutations( bin_sort, other_views... );
+}
+
 template <typename DeviceType>
+template <typename View, typename... OtherViews>
 void DistributedSearchTreeImpl<DeviceType>::sortResults(
-    Kokkos::View<int *, DeviceType> query_ids,
-    Kokkos::View<int *, DeviceType> results,
-    Kokkos::View<int *, DeviceType> ranks,
-    Kokkos::View<double *, DeviceType> *distances_ptr )
+    View keys, OtherViews... other_views )
 {
-    int const n = query_ids.extent( 0 );
+    auto const n = keys.extent( 0 );
     // If they were no queries, min_val and max_val values won't change after
     // the parallel reduce (they are initialized to +infty and -infty
     // respectively) and the sort will hang.
     if ( n == 0 )
         return;
 
-    typedef Kokkos::BinOp1D<Kokkos::View<int *, DeviceType>> CompType;
+    using Comp = Kokkos::BinOp1D<View>;
+    using Value = typename View::non_const_value_type;
 
-    Kokkos::Experimental::MinMaxScalar<int> result;
-    Kokkos::Experimental::MinMax<int> reducer( result );
-    parallel_reduce(
-        Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
-        Kokkos::Impl::min_max_functor<Kokkos::View<int *, DeviceType>>(
-            query_ids ),
-        reducer );
+    Kokkos::Experimental::MinMaxScalar<Value> result;
+    Kokkos::Experimental::MinMax<Value> reducer( result );
+    parallel_reduce( Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
+                     Kokkos::Impl::min_max_functor<View>( keys ), reducer );
     if ( result.min_val == result.max_val )
         return;
-    Kokkos::BinSort<Kokkos::View<int *, DeviceType>, CompType> bin_sort(
-        query_ids, CompType( n / 2, result.min_val, result.max_val ), true );
+    Kokkos::BinSort<View, Comp> bin_sort(
+        keys, Comp( n / 2, result.min_val, result.max_val ), true );
     bin_sort.create_permute_vector();
-    bin_sort.sort( results );
-    bin_sort.sort( ranks );
-    if ( distances_ptr )
```

Comment:
Why don't you need to check this anymore?

Reply:
Because now I pass the view rather than a pointer to it.  The old signature with a default value `nullptr` for the pointer was a hack to not implement two version of `sortResults()`.